modules/mig/manifests/agent/disable.pp

```diff
@@ -0,0 +1,64 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+class mig::agent::disable {
+
+    $mig_path = $::operatingsystem ? { /(CentOS|Ubuntu)/ => '/sbin/mig-agent', Darwin => '/usr/local/bin/mig-agent' }
+    case $::operatingsystem {
+        # We want to disable mig-agent on Linux and MacOS workers
+        Ubuntu: {
+            case $::operatingsystemrelease {
+                16.04: {
+                    # Kill mig agent service if mig-agent is running
+                    exec {
+                        'kill mig':
+                            command   => "/bin/kill -9 $(${mig_path} -q=pid)",
+                            subscribe => Package['mig-agent'],
+                            notify    => Service['mig-agent'],
+                            onlyif    => "/usr/bin/test `mig-agent -q=pid|wc -l` -eq 1"
```

Comment:
If mig has been removed from a previous run, how are you going exec mig to determine the pid in the onlyif test?